flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/aggfunctions/MaxAggFunction.scala

```diff
@@ -28,8 +28,9 @@ import org.apache.flink.table.functions.{Accumulator, AggregateFunction}
   * @tparam T the type for the aggregation result
   */
 abstract class MaxAggFunction[T](implicit ord: Ordering[T]) extends AggregateFunction[T] {
+
   /** The initial accumulator for Max aggregate function */
-  class MaxAccumulator[T] extends Accumulator {
+  class MaxAccumulator[T] extends JTuple2[T, Boolean] with Accumulator {
     var max: T = null.asInstanceOf[T]
```

Comment:
Instead of adding a field here, you need to provide default values in the `createAccumulator` method. Note, `f0`  and `f1` may not be null. Otherwise, the tuple cannot be serialized.